plugins/modules/katello_activation_key.py

```diff
@@ -145,88 +114,151 @@ def override_to_boolnone(override):
 
 
 def main():
-    module = KatelloEntityAnsibleModule(
-        argument_spec=dict(
+    module = KatelloEntityApypieAnsibleModule(
+        entity_spec=dict(
             name=dict(required=True),
             new_name=dict(),
-            lifecycle_environment=dict(),
-            content_view=dict(),
-            subscriptions=dict(type='list'),
-            host_collections=dict(type='list'),
-            content_overrides=dict(type='list'),
-            auto_attach=dict(type='bool', default=True),
+            lifecycle_environment=dict(type='entity', flat_name='environment_id'),
+            content_view=dict(type='entity', flat_name='content_view_id'),
+            host_collections=dict(type='entity_list', flat_name='host_collection_ids'),
+            auto_attach=dict(type='bool'),
+        ),
+        argument_spec=dict(
+            subscriptions=dict(type='list', elements='dict', options=dict(
+                name=dict(required=True),
+            )),
+            content_overrides=dict(type='list', elements='dict', options=dict(
+                label=dict(required=True),
+                override=dict(choises=['enabled', 'disabled']),
+            )),
             state=dict(default='present', choices=['present', 'present_with_defaults', 'absent', 'copied']),
         ),
         required_if=[
             ['state', 'copied', ['new_name']],
         ],
     )
 
-    (entity_dict, state) = module.parse_params()
+    entity_dict = module.clean_params()
 
     module.connect()
 
-    entity_dict['organization'] = find_organization(module, name=entity_dict['organization'])
-    if 'lifecycle_environment' in entity_dict:
-        entity_dict['lifecycle_environment'] = find_lifecycle_environment(module, entity_dict['lifecycle_environment'], entity_dict['organization'])
+    # TODO move to foreman_helper
+    module.foremanapi.apidoc['docs']['resources']['subscriptions']['methods'][0]['params'][7]['required'] = False
```

Comment:
If i have not overlooked something, this is the WIP in this PR.

Reply:
What is this? method show, param org_id?

Reply:
Index and org_id

Reply:
index, oups :)